### PR TITLE
Add file view by week, fixes #14

### LINKED ISSUE #14: Produce file view by percentage of students in the last X weeks
Need to generate a similar view of file viewing patterns as demoed in this [Tableau view.](https://tableau.dsc.umich.edu/#/site/UM/views/Aug1-Fileleaderboard-STATS250SP18/Dashboard1?:iid=1)

### PR DESCRIPTION
**Pull request by zqian**
Added the file view by week data.

### CODE REVIEW

**Comment by jonespm** [comment]
You can have PR's close issues with https://help.github.com/articles/closing-issues-using-keywords/

I also don't know if our team uses Zenhub but it also is really nice (and free for open projects)

File: dashboard/cron.py [hunk 1]
```diff
@@ -118,28 +111,25 @@ def update_with_udw_access(request):
...
+                                'datetime(EVENT_TIME) as ACCESS_TIME ' \
                                 'FROM learning_datasets.enriched_events ' \
                                 'where JSON_EXTRACT_SCALAR(event, "$.edApp.id") = \'http://umich.instructure.com/\' ' \
                                 'and event_type = \'NavigationEvent\' ' \
                                 'and JSON_EXTRACT_SCALAR(event, "$.object.name") = \'attachment\' ' \
                                 'and JSON_EXTRACT_SCALAR(event, "$.action") = \'NavigatedTo\' ' \
                                 'and JSON_EXTRACT_SCALAR(event, "$.membership.member.id") is not null ' \
                                 'and SUBSTR(JSON_EXTRACT_SCALAR(event, "$.group.id"),31) = \'' + UDW_COURSE_ID + '\' '
-                        logger.debug(query)
+                        logger.info(query)
```
Comment (jonespm) [inline dashboard/cron.py]: I'd probably move these back to debug, was that what you were talking about?

File: dashboard/static/js/file_access_within_week.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
+var highlightData = function(file_name) {
```
Comment (jonespm) [inline dashboard/static/js/file_access_within_week.js]: We really should namespace these javascript functions perhaps with revealing module. If you'd like create another ticket and I can clean this up. ;)

 https://addyosmani.com/resources/essentialjsdesignpatterns/book/#revealingmodulepatternjavascript

File: dashboard/views.py [hunk 3]
```diff
@@ -63,6 +64,9 @@ def home(request):
 def files(request):
     return render_to_response("files.html")
 
+def view_file_access_within_week(request):
```
Comment (jonespm) [inline dashboard/views.py]: This looks like it could just use TemplateView in the urls

File: dashboard/urls.py [hunk 4]
```diff
@@ -34,15 +34,17 @@
...
+    url(r'^view_file_access_within_week', views.view_file_access_within_week, name='view_file_access_within_week'),
     url(r'^grade_distribution', views.grade_distribution, name='grade_distribution'),
 
     # load file information
     url(r'^load_data', views.load_data, name='load_data'),
 
     # load data from UDW
     url(r'^update_with_udw_file', cron.update_with_udw_file, name='update_with_udw_file'),
-    url(r'^update_with_udw_access', cron.update_with_udw_access, name='update_with_udw_access'),
+    url(r'^update_with_bq_access', cron.update_with_bq_access, name='update_with_bq_access'),
```
Comment (jonespm) [inline dashboard/urls.py]: Can you update this in the testloader.html?

**Comment by jonespm** [comment]
Added a few inline comments, would be nice to append on-top of this. Love the new views!

File: dashboard/cron.py (hunk 1 above)
Comment (zqian) [reply dashboard/cron.py]: Yes. Changed it to debug.

File: dashboard/static/js/file_access_within_week.js (hunk 2 above)
Comment (zqian) [reply dashboard/static/js/file_access_within_week.js]: Another ticket would be good once JS files are all done.

File: dashboard/urls.py (hunk 4 above)
Comment (zqian) [reply dashboard/urls.py]: Done!

File: dashboard/views.py (hunk 3 above)
Comment (zqian) [reply dashboard/views.py]: Done!

**Comment by jonespm** [comment]
Codacy has a lot of false positives when it first hooks in, we'll clean that up.

**Comment by zqian** [comment]
I have fixed the sql parameters for BigQuery and MySQL. I did not have time to get the postgresql ones right.

**Pull request merged**